lib/rollup-funnel.js

```diff
@@ -45,25 +47,95 @@ function RollupFunnel(inputNodes, options) {
     throw new Error('Must specify exactly one of `include` or `exclude`.');
   }
 
-  this.originalInput = inputNodes;
   options = options || {};
+
+  this._depGraph = null;
+  this._depGraphTree = null;
+
   Plugin.call(this, [inputNodes], {
-    annotation: options.annotation
+    annotation: options.annotation,
+    persistentOutput: true
   });
   this.options = options;
 }
 
 RollupFunnel.prototype._copy = Funnel.prototype._copy;
 
+/**
+ * Constructs an FSTree from the paths stored in _depGraph.
+ */
+RollupFunnel.prototype._getFSTree = function(paths) {
+  var inputPath = this.inputPaths[0];
+  var entries = paths.map(function(entryPath) {
+    var absolutePath = path.join(inputPath, entryPath);
+
+    if (!fs.existsSync(absolutePath)) {
+      return;
+    }
+
+    var stat = fs.lstatSync(absolutePath);
+    var isDirectory = stat.isDirectory();
+
+    return {
```

Comment:
I would appreciate this, or even having `fromPaths` do the stating. It was super unintuitive to me why `fromPaths` wasn't doing what I wanted it to do.